src/dataflow/src/render/mod.rs

```diff
@@ -425,15 +401,10 @@ where
                     ArrangementFlavor::Trace(idx_id, ok_arranged, err_arranged),
                 ),
             );
-            tokens
-                .additional_tokens
-                .entry(idx_id)
-                .or_insert_with(Vec::new)
-                .push(Rc::new((
-                    ok_button.press_on_drop(),
-                    err_button.press_on_drop(),
-                    token,
-                )));
+            tokens.insert(
+                idx_id,
+                Rc::new((ok_button.press_on_drop(), err_button.press_on_drop(), token)),
+            );
```

Comment:
This is a moment where I'm not sure exactly what is going on, and it might need an assert. It looks like the prior implementation would extend the entry for `idx_id` rather than insert it, which I think corresponds to inserting the same index multiple times. I think doing that should be a bug, but maybe I'm missing something. In either case, maybe this calls for an assert so that we don't silently do something different and risk correctness issues.

Reply:
This is only called with unique keys, as it results from the keys of a `BTreeMap`. So, not a regression afaict, and also not a lower level of invariant checking than we have elsewhere, so comfortable leaving it as is.